plan/physical_plan_builder.go

```diff
@@ -783,6 +785,178 @@ func (p *Union) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanInfo,
 	return info, nil
 }
 
+// The first return value is the new expression, the second is a bool value tell whether the below expression is all constant.
+// The Second is used for simplify the scalar function.
+// If the args of one scalar function are all constant, we will substitute it to constant.
+func substituteCorCol2Constant(cond expression.Expression) expression.Expression {
```

Comment:
One is change correlated column to expression.One, another is eval the correlated column to Datum. I thought eval a correlated column when its Data is nothing doesn't make sense.